week1/utilities/data_prepper.py

```diff
@@ -236,22 +236,36 @@ def __log_ltr_query_features(self, query_id, key, query_doc_ids, click_prior_que
                                                 size=len(query_doc_ids), terms_field=terms_field)
         ##### Step Extract LTR Logged Features:
         # IMPLEMENT_START --
-        print("IMPLEMENT ME: __log_ltr_query_features: Extract log features out of the LTR:EXT response and place in a data frame")
+        #print("IMPLEMENT ME: __log_ltr_query_features: Extract log features out of the LTR:EXT response and place in a data frame")
         # Loop over the hits structure returned by running `log_query` and then extract out the features from the response per query_id and doc id.  Also capture and return all query/doc pairs that didn't return features
         # Your structure should look like the data frame below
-        feature_results = {}
-        feature_results["doc_id"] = []  # capture the doc id so we can join later
-        feature_results["query_id"] = []  # ^^^
-        feature_results["sku"] = []
-        feature_results["name_match"] = []
-        rng = np.random.default_rng(12345)
-        for doc_id in query_doc_ids:
-            feature_results["doc_id"].append(doc_id)  # capture the doc id so we can join later
-            feature_results["query_id"].append(query_id)
-            feature_results["sku"].append(doc_id)  
-            feature_results["name_match"].append(rng.random())
-        frame = pd.DataFrame(feature_results)
-        return frame.astype({'doc_id': 'int64', 'query_id': 'int64', 'sku': 'int64'})
+
+        response = self.opensearch.search(body=log_query, index=self.index_name)
+        if response and response['hits']['hits'] and len(response['hits']['hits']) > 0:
+            hits = response['hits']['hits']
+            
+            # extract the features based on the features passed into the Prepper
+            feature_results = {}
+            feature_results["doc_id"] = []  # capture the doc id so we can join later
+            feature_results["query_id"] = []  # ^^^
+            feature_results["sku"] = []
+
+            for hit in hits:
+                feature_results["doc_id"].append(hit["_id"])
+                feature_results["sku"].append(hit["_source"]["sku"][0])
+                feature_results["query_id"].append(query_id)
+                log_entries = hit["fields"]["_ltrlog"][0]["log_entry"]
+                for entry in log_entries:
+                    feature_results_name_value = feature_results.get(entry.get("name"))
+                    if feature_results_name_value is None:
+                        feature_results_name_value = []
+                        feature_results[entry.get("name")] = feature_results_name_value
+                    feature_results_name_value.append(entry.get("value", 0))
```

Comment:
Suggestion: I think this part can be simplified e.g

```python
 feature_results[entry.get("name")] = [(entry.get("value", 0)]

Reply:
ok